src/test/java/WebDriverScenarioWithTests.java

```diff
@@ -0,0 +1,129 @@
+        WebElement searchInput = driver.findElement(By.name("q"));
+        searchInput.sendKeys("wikipedia");
+        new WebDriverWait(driver, 20).until(ExpectedConditions
+                .elementToBeClickable(By.xpath("//input[@name='btnK']"))).click();
+        wait.until(pageLoadCondition);
+        List<WebElement> searchResults = driver.findElements(By.xpath("//cite[contains(text(), 'wikipedia')]"));
+        logger.info(String.format("Search results number for requested text: %s", searchResults.size()));
+        Assert.assertFalse(searchResults.contains("wikipedia"), "Search results do not contain expected text.");
+
+        logger.info("3. Validate that first found link is related to wikipedia.");
+        WebElement firstElementOfSearchResults = driver.findElement(By.xpath("//cite[@class='iUh30 Zu0yb qLRx3b tjvcx']"));
+        String contentOfFirstSearchResult = firstElementOfSearchResults.getText();
+        new WebDriverWait(driver, 50).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[contains(text(),'wikipedia')]")));
+        logger.info(String.format("First element of search results is %s", contentOfFirstSearchResult));
+        wait.until(pageLoadCondition);
+        Assert.assertTrue(contentOfFirstSearchResult.contains("wikipedia"), "First element of search results do not contain expected text.");
+
+        logger.info("4. Navigate to https://en.wikipedia.org via first found link.");
+        WebElement firstEnglishWikipediaLink = driver.findElement(By.xpath("//cite[text()='https://en.wikipedia.org']"));
+        new WebDriverWait(driver, 30).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[text()='https://en.wikipedia.org']")));
+        String contentOfFirstEnglishWikipediaLink = firstEnglishWikipediaLink.getText();
+        firstEnglishWikipediaLink.click();
+        wait.until(pageLoadCondition);
+
+        logger.info("5. Assert that we are currently on wikipedia page.");
+        logger.info(String.format("The link clicked is %s", contentOfFirstEnglishWikipediaLink));
+
+        Assert.assertTrue(contentOfFirstEnglishWikipediaLink.contains("wikipedia"), "The element is clicked do not contains expected text.");
+
+        logger.info("6. Get Page Title name and Title length -> Print Page Title and Title length on the Console via logger.");
+        String titleOfClickedEnglishWikipediaLink = driver.getTitle();
+        int titleLength = titleOfClickedEnglishWikipediaLink.length();
+        logger.info(String.format("The title of clicked link is ", titleOfClickedEnglishWikipediaLink, ".\n Title length is ", titleLength, " symbols."));
+
+        logger.info("7. Get Page URL and verify if the it is a correct page opened.");
+        String clickedURL = driver.getCurrentUrl();
+
+        Assert.assertTrue(clickedURL.contains("wikipedia"), "The opened link do not content expected text.");
```

Comment:
The opened link do not contAIN expected text.

Reply:
My apologizes, corrected.